### PR TITLE
1010CG Cypress Testing

### PR DESCRIPTION
**Pull request by LinnJS**
## Description
Create Cypress e2e test for 1010CG

## Testing done
Cypress test

## Screenshots
image

## Acceptance criteria
- [x] Created automated e2e Cypress test

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/caregivers/containers/App.jsx [hunk 1]
```diff
@@ -15,6 +15,8 @@ const App = ({ loading, isFormAvailable, location, children }) => {
     return <LoadingIndicator />;
   }
 
+  console.log('isFormAvailable', isFormAvailable);
```
Comment (LinnJS) [inline src/applications/caregivers/containers/App.jsx:18]: Just for troubleshooting, need to remove

File: src/applications/vaos/containers/ExpressCareInfo.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import React, { Component } from 'react';
```
Comment (LinnJS) [inline src/applications/vaos/containers/ExpressCareInfo.jsx:1]: Do not know how this got here, need to reset this file

File: src/applications/caregivers/tests/e2e/1010CG-secondaryTwo.cypress.spec.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    pageHooks: {
+      introduction: () => {
+        // Hit the start button
+        cy.findAllByText(/start/i, { selector: 'button' })
+          .first()
+          .click();
+
+        // Click past the ITF message
+        cy.findByText(/continue/i, { selector: 'button' }).click();
+      },
```
Comment (U-DON) [inline src/applications/caregivers/tests/e2e/1010CG-secondaryTwo.cypress.spec.js:36]: Keep in mind this will have to be updated when #13740 gets merged.

**Review by U-DON — COMMENTED** [review]
Is there a reason you needed to split the form tests? If it's due to testing the review page differently,  you can [test it conditionally](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/testing/e2e/cypress/support/form-tester/README.md#testkey) with `testKey` or `testData`.

**Comment by LinnJS** [comment]
@U-DON Implemented feedback, using `testKeys`!

File: src/applications/caregivers/tests/e2e/1010cg.cypress.spec.js [hunk 4]
```diff
@@ -0,0 +1,164 @@
+import path from 'path';
+
+import testForm from 'platform/testing/e2e/cypress/support/form-tester';
+import { createTestConfig } from 'platform/testing/e2e/cypress/support/form-tester/utilities';
+
+import formConfig from '../../config/form';
+import manifest from '../../manifest.json';
+
+const veteranLabel = `Enter Veteran's or service member\u2019s full name`;
```
Comment (Mottie) [inline src/applications/caregivers/tests/e2e/1010cg.cypress.spec.js:9]: `'Enter Veteran\u2019s ...'`?

**Review by Mottie — APPROVED** [review]

File: src/applications/caregivers/tests/e2e/1010cg.cypress.spec.js (hunk 4 above)
Comment (LinnJS) [reply src/applications/caregivers/tests/e2e/1010cg.cypress.spec.js]: Yeah this is not ideal, using this as the labels for each field, have another ticket to refactor this as well as the switch statement inside the checkbox components.

File: src/applications/caregivers/tests/e2e/1010cg.cypress.spec.js [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+        cy.findAllByText(/start/i, { selector: 'button' })
+          .first()
+          .click();
+      },
+      'review-and-submit': () => {
+        cy.get('@testKey').then(testKey => {
+          switch (testKey) {
+            case 'oneSecondaryCaregivers.json':
+              // sign signature as veteran
+              cy.get(`[data-test-id="${veteranLabel}-signature-input"]`)
```
Comment (U-DON) [inline src/applications/caregivers/tests/e2e/1010cg.cypress.spec.js:43]: Will [`findByTestId`](https://testing-library.com/docs/dom-testing-library/api-queries#bytestid) work for these?

More importantly, the queries in the DOM testing library are [listed in order of recommendation](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/testing/end-to-end/cypress-migration-guide.md#interacting-with-page-elements-). So if there's a label that goes with this input, it's usually better to match by that.

File: src/applications/caregivers/tests/e2e/1010cg.cypress.spec.js [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+const secondaryTwoLabel =
+  'Enter Secondary Family Caregiver\u2019s (2) full name';
+
+const testSecondaryTwo = createTestConfig(
+  {
+    dataPrefix: 'data',
+    dataSets: [
+      'requiredOnly.json',
+      'oneSecondaryCaregivers.json',
+      'twoSecondaryCaregivers.json',
```
Comment (U-DON) [inline src/applications/caregivers/tests/e2e/1010cg.cypress.spec.js:21]: Feel free  to omit the `.json`.

File: src/applications/caregivers/tests/e2e/1010cg.cypress.spec.js (hunk 5 above)
Comment (LinnJS) [reply src/applications/caregivers/tests/e2e/1010cg.cypress.spec.js]: I meant to use this, refactored and code is much cleaner and more readable, great catch thank you @U-DON!

**Review by bradl3yC — APPROVED** [review]

**Pull request merged**